security/src/main/java/io/airlift/security/utils/KerberosUtil.java

```diff
@@ -0,0 +1,173 @@
+package io.airlift.security.utils;
+
+import com.google.common.base.Throwables;
+import io.airlift.log.Logger;
+import io.airlift.security.exception.AuthenticationException;
+import org.ietf.jgss.GSSContext;
+import org.ietf.jgss.GSSCredential;
+import org.ietf.jgss.GSSException;
+import org.ietf.jgss.GSSManager;
+import org.ietf.jgss.GSSName;
+import org.ietf.jgss.Oid;
+
+import javax.security.auth.Subject;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.security.Principal;
+import java.security.PrivilegedAction;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public class KerberosUtil
+{
+    private static final String KRB5_LOGIN_MODULE_NAME = "com.sun.security.auth.module.Krb5LoginModule";
+    private static final Oid SPNEGO_OID;
+    private static final Oid KERBEROS_OID;
+    private static final Oid[] SUPPORTED_OIDS;
+
+    static {
+        try {
+            SPNEGO_OID = new Oid("1.3.6.1.5.5.2");
+            KERBEROS_OID = new Oid("1.2.840.113554.1.2.2");
+            SUPPORTED_OIDS = new Oid[] {SPNEGO_OID, KERBEROS_OID};
+        }
+        catch (GSSException e) {
+            throw new AuthenticationException(e);
```

Comment:
This should be an assertion error. It means that either there's a bug in the code (a typo in the OID string) or the JVM is broken.